### PR TITLE
feat(locale): add en_HK locale

### PR DESCRIPTION
**Pull request by matthewmayer**
Adds a locale for English (Hong Kong)

A few points of note:
* Hong Kong has no ZIP codes
* Override surnames but fall back to en first names
* Street names are generated with a mix of English and Cantonese parts

### CODE REVIEW

File: src/locales/en_HK/location/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import street_english_part from './street_english_part';
+import street_pattern from './street_pattern';
+import street_prefix from './street_prefix';
+import street_suffix from './street_suffix';
+
+const location: LocationDefinitions = {
+  building_number,
+  city,
+  default_country,
+  postcode,
```
Comment (ST-DDT) [inline src/locales/en_HK/location/index.ts:23]: Do we want to also explicitly set `postcode_by_state` to null?

Comment (matthewmayer) [reply src/locales/en_HK/location/index.ts:23]: Sure

**Comment by ST-DDT** [comment]
Note to self: Update migration guide to use post codes: https://github.com/faker-js/faker/pull/2004#discussion_r1174588696 once this is merged.

**Review by Shinigami92 — DISMISSED** [review]
already give my approval, but looks like I need to re-approve again later

File: src/locales/en_HK/location/city.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
+export default [
```
Comment (ST-DDT) [inline src/locales/en_HK/location/city.ts:1]: According to Wikipedia this refers to a city part of Hong Kong.
But this sounds like an OK solution for a "city state".

**Review by ST-DDT — APPROVED** [review]

**Comment by ST-DDT** [comment]
> Note to self: Update migration guide to use post codes: [#2004 (comment)](https://github.com/faker-js/faker/pull/2004#discussion_r1174588696) once this is merged.

> @matthewmayer [update docs](https://github.com/faker-js/faker/pull/2083/commits/9aedcd6b6d379efa5eb1c56c2bf277a0ca216cb9)

My original intention was to do that in a separate PR, but this works as well. Thanks for your help! 👍

**Review by Shinigami92 — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

File: src/locales/en_HK/location/city.ts (hunk 2 above)
Comment (matthewmayer) [reply src/locales/en_HK/location/city.ts:1]: yes, i tried to map city and state in the most logical way
A) city => villages/districts/islands of Hong Kong
B) state => the three main constituent parts of Hong Kong

HK addresses are generally written as street name, "city", "state" using these definitions.

**Pull request merged**